### DISCUSSION TITLE
Project Folder Structure.

### DISCUSSION THREAD

**Discussion by kaustubhn** [P0W/option_strategies#3] [Ideas]
Proposal to redo (organise) the code base so that it can better handle scale, production etc.

Also, help manage the codebase as it is growing.

Proposed folder structure as below:
.
├── Makefile
├── README.md
├── compose
├── logs
 │   └── daily_logs_new.txt
├── requirements.txt
├── resources
 │   ├── backtest
 │   │   └── algotest_strategy_short_strangle_nifty.pdf
 │   └── options_strategies.png
├── src
 │   ├── DailyShorts
 │   │   ├── __init__.py
 │   │   ├── function.json
 │   │   ├── readme.md
 │   │   └── sample.dat
 │   ├── bot
 │   │   └── bot.py
 │   ├── clients
 │   │   ├── client_5paisa.py
 │   │   ├── client_dummy.py
 │   │   ├── client_fyers.py
 │   │   ├── client_shoonya.py
 │   │   ├── client_upstock.py
 │   │   ├── client_zerodha.py
 │   │   └── iclientmanager.py
 │   ├── common
 │   │   ├── __init__.py
 │   │   ├── live_feed_manager.py
 │   │   ├── order_manager.py
 │   │   └── strikes_manager.py
 │   ├── daily_short.py
 │   ├── simulator
 │   │   ├── docker-compose.yml
 │   │   └── feeds.py
 │   ├── strategy
 │   │   ├── __init__.py
 │   │   └── base_strategy.py
 │   └── utils.py
└── tests
    ├── live_feed_manager_test.py
    └── strategy_test.py

**Comment by P0W** [comment]
Apart from resources I think I have what you proposed. Let me know if I missed something. I can move the resources into a folder, if that you increase maintainability

**Reply by kaustubhn** [reply]
You do, but I think in the above structure, the fiels you have are in the src folder, also I have docs & compose for documentation & docker compose file, also I have Makefile, that enables us to write commands and run them quickly.

Also if we follow that approach we will also add .envs folder for different environments as well.

**Reply by kaustubhn** [reply]
You can refer to this branch: [here](https://github.com/kaustubhn/option_strategies/tree/feature/refactor)
